src/db/models/JobModel.ts

```diff
@@ -177,4 +198,78 @@ export class JobModel extends Model<JobModel> {
     onDelete: "CASCADE",
   })
   applications: ApplicationModel[];
+
+  @AfterCreate
+  static async sendEmailHook(instance: JobModel) {
+    console.log("New entry created");
+    const mailerService = new EmailService();
+
+    const dto: SendEmailDto = {
+      from: { name: "TPC Portal", address: "aryangkulkarni@gmail.com" },
+      recepients: [{ address: "me210003016@iiti.ac.in" }], // Put your email address for testing
+      subject: "Test email",
+      html: "<p>New job entry was created</p>",
+    };
+
+    // Send email
+    await mailerService.sendEmail(dto);
+  }
+
+  @BeforeBulkUpdate
+  static async sendEmailOnEventChange(options: any) {
+    const jobs = await JobModel.findAll({ where: options.where });
+    // console.log("Before Update Called");
+    for (const job of jobs) {
+      const previousActive = job.active;
+      const newActive = options.attributes.active;
+
+      if (previousActive === false && newActive === true) {
+        // console.log("Before Update - Sending Email for Job ID: ", job.id);
+
+        const mailerService = new EmailService();
+
+        const students: StudentModel[] = await StudentModel.findAll();
+        const salaries = await SalaryModel.findAll({
+          where: {
+            jobId: job.id,
+          },
+        });
+        const eligibleStudents = new Set<StudentModel>();
+        console.log("salaries", salaries);
+
+        // Iterate over each salary and filter students
+        for (const salary of salaries) {
+          students.forEach((student) => {
+            if (
+              student.cpi >= salary.minCPI &&
+              salary.programs.includes(student.programId) &&
+              salary.categories.includes(student.category) &&
+              salary.genders.includes(student.gender) &&
+              student.tenthMarks >= salary.tenthMarks &&
+              student.twelthMarks >= salary.twelthMarks
+            ) {
+              console.log("student added", student.id);
+              eligibleStudents.add(student);
+            }
+          });
+        }
```

Comment:
use db query instead, more efficient.
you can just use `filterSalaries` which does that.
use case here https://github.com/Web-Team-IITI-Gymkhana/tpc-backend/blob/3a80c7e470c220c2bb35ad61cba93c6e342b3b5b/src/student-view/salary/salary.service.ts#L51

also include this condition
https://github.com/Web-Team-IITI-Gymkhana/tpc-backend/blob/3a80c7e470c220c2bb35ad61cba93c6e342b3b5b/src/student-view/salary/salary.service.ts#L64-L68